specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/integrations.json

```diff
@@ -0,0 +1,653 @@
+            "type": "string"
+          },
+          "x-ms-identifiers": []
+        },
+        "includeMemberScopes": {
+          "type": "string",
+          "description": "Defines whether the integration is management scope integration (master connector as a single scope or management scope)"
+        },
+        "metadata": {
+          "$ref": "#/definitions/IntegrationMetadata"
+        },
+        "integrationAdditionalData": {
+          "type": "object",
+          "description": "Additional data",
+          "$ref": "#/definitions/IntegrationAdditionalData"
+        }
+      },
+      "required": [
+        "displayName",
+        "integrationType"
+      ]
+    },
+    "IntegrationRootDetails": {
+      "type": "object",
+      "description": "The integration root details",
+      "properties": {
+        "resourceId": {
+          "description": "The integration root resource Id",
+          "type": "string",
+          "readOnly": true
+        },
+        "distance": {
+          "description": "The number of nodes between integration and the root integration resource",
+          "type": "number",
+          "readOnly": true
+        }
+      }
+    },
+    "IntegrationMetadata": {
+      "type": "object",
+      "description": "The integration metadata",
+      "properties": {
+        "createdBy": {
+          "description": "Integration Created by object id (GUID)",
+          "type": "string",
+          "readOnly": true
+        },
+        "createdOn": {
+          "description": "Integration creation date",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "updatedBy": {
+          "description": "Integration last updated by object id (GUID)",
+          "type": "string",
+          "readOnly": true
+        },
+        "updatedOn": {
+          "description": "Integration last update date",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        }
+      }
+    },
+    "IntegrationAdditionalData": {
+      "type": "object",
+      "description": "The integration additional data (abstract)",
+      "discriminator": "integrationPlatform",
+      "properties": {
+        "integrationPlatform": {
+          "type": "string",
+          "description": "Describes the integration platform"
+        }
+      },
+      "required": [
+        "integrationPlatform"
+      ]
+    },
+    "ServiceNowAdditionalData": {
+      "description": "Service Now specific additional data",
+      "allOf": [
+        {
+          "$ref": "#/definitions/IntegrationAdditionalData"
+        },
+        {
+          "properties": {
+            "instanceUrl": {
+              "description": "Integration creation date",
+              "type": "string"
+            },
+            "clientId": {
+              "description": "Integration creation date",
+              "type": "string"
+            },
+            "clientSecret": {
```

Comment:
>clientSecret [](http://example.com/codeflow?start=13&length=12)

Please ensure that your service implementation doesn't return any secrets such as this, in the GET/PUT/PATCH response. Only POST Action can be used to return secret values if needed.